src/include/processor/operator/persistent/reader/parquet/parquet_reader.h

```diff
@@ -0,0 +1,80 @@
+#pragma once
+
+#include "column_reader.h"
+#include "common/data_chunk/data_chunk.h"
+#include "common/types/types.h"
+#include "parquet/parquet_types.h"
+#include "resizable_buffer.h"
+#include "struct_column_reader.h"
+#include "thrift_tools.h"
+
+namespace kuzu {
+namespace processor {
+
+struct ParquetReaderPrefetchConfig {
+    // Percentage of data in a row group span that should be scanned for enabling whole group
+    // prefetch
+    static constexpr double WHOLE_GROUP_PREFETCH_MINIMUM_SCAN = 0.95;
+};
+
+struct ParquetReaderScanState {
+    std::vector<uint64_t> groupIdxList;
+    int64_t currentGroup = -1;
+    uint64_t groupOffset;
+    std::unique_ptr<common::FileInfo> fileInfo;
+    std::unique_ptr<ColumnReader> rootReader;
+    std::unique_ptr<kuzu_apache::thrift::protocol::TProtocol> thriftFileProto;
+
+    bool finished;
+
+    ResizeableBuffer defineBuf;
+    ResizeableBuffer repeatBuf;
+
+    bool prefetchMode = false;
```

Comment:
Add a comment to explain `prefetchMode` here, and comment on we're setting it to false always.
I would also consider opening an issue to clean this up or revisit once we add remote fs, in case we forget later and switched to another approach of handling this for reading from remote fs.